### PR TITLE
Pass app info status

### PR DESCRIPTION
**Pull request by JettTech**
### Updates:
 - returns app-info status back in hhdt package
 - add app-info status filters for error checks
 - adds app-info status test
 - bumps `mock-holochain-conductor` to v0.3.2
 - updates tests to use unique users for each e2e sign-up/sign-in test

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -449,16 +449,31 @@ class Envoy {
         if (!appInfo) {
           log.error("Conductor call 'appInfo' returned non-success response: %s", appInfo);
           throw new HoloError(`Failed to call 'appInfo' for installed_app_id'${installed_app_id}'.`);
+        } else if (appInfo.status!.inactive) {
+          log.error("Conductor call 'appInfo' returned as inactive because: %s", JSON.stringify(appInfo.status!.inactive));
+          if (appInfo.status!.inactive!.reason!.quarantined) {
```
Comment (timotree3) [inline src/index.ts:505]: Is there a reason to add these `!`s?

File: tests/e2e/test_chaperone.js [hunk 2]
```diff
@@ -126,6 +126,72 @@ describe("Server", () => {
     await closedPromise
   })
 
+  it.only("should return correct appInfo status after app is deactivated", async function() {
```
Comment (timotree3) [inline tests/e2e/test_chaperone.js]: remove .only

Comment (timotree3) [inline tests/e2e/test_chaperone.js]: Am I missing something? This test doesn't seem to call appInfo (not directly at least)

File: package.json [hunk 3]
```diff
@@ -28,7 +28,7 @@
   "dependencies": {
     "@holo-host/cryptolib": "^0.3.0",
     "@holo-host/data-translator": "^0.1.1",
-    "@holochain/conductor-api": "holochain/holochain-conductor-api#34acd25ed232bbbaecac8e27c26fcfc4db77a557",
+    "@holochain/conductor-api": "file:/home/liza/Documents/gitrepos/holochain/holochain-conductor-api",
```
Comment (timotree3) [inline package.json]: Fix these two file:/ deps before merging

**Review by timotree3 — COMMENTED** [review]
I had a couple of questions and there's a couple WIP things that need to be tidied before merging. Maybe I'm reviewing this too early :)

File: tests/e2e/test_chaperone.js (hunk 2 above)
Comment (JettTech) [reply tests/e2e/test_chaperone.js]: Test is ready now. :)

File: tests/e2e/test_chaperone.js [hunk 4]
```diff
@@ -135,6 +135,79 @@ describe("Server", () => {
...
+    log.info("Completed signup response: %s", signupResponse);
+    expect(signupResponse).to.equal(true);
+    expect(signoutResponse).to.equal(true); 
+
+    // Delay is added so that the signout calls have time to finish
+    await delay(15000);
+
+    // 2. Then make appInfo call using the agent   pubkey (since the agent has signed out and closed their ws, their cell should have been deactivated and the app should return as 'inactive')
+    // cannot make call from within chaperone instance bc the getAppInfo call doesn't accept provided a `installed_app_id`
+    const { Client: RPCWebsocketClient } = require('rpc-websockets')
```
Comment (timotree3) [inline tests/e2e/test_chaperone.js]: Can this require go at the top of the file?

**Review by timotree3 — APPROVED** [review]
Looks good besides one import and that tests are failing

File: src/index.ts (hunk 1 above)
Comment (JettTech) [reply src/index.ts:505]: In TS it acts as a non-null assertion operator and tells the compiler that the marked expression or obj cannot/will not be null or undefined, which removes the possibility of complaining about it being null or undefined.  Overall, I just find if more explicit.
https://github.com/Microsoft/TypeScript/wiki/What's-new-in-TypeScript#non-null-assertion-operator

**Review by timotree3 — APPROVED** [review]

**Pull request merged**